### ISSUE TITLE
Keeps crashing on me: Methods marked with @UiThread must be executed on the main thread

### ISSUE THREAD

**Issue by Tomucha** [ufukhawk/safe_device#1]
I don't see what might be wrong on my side ...

It's just:
bool canMockLocation = await SafeDevice.canMockLocation;

```
E/AndroidRuntime( 7592): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: Timer-0
E/AndroidRuntime( 7592): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:1230)
E/AndroidRuntime( 7592): 	at io.flutter.embedding.engine.FlutterJNI.invokePlatformMessageResponseCallback(FlutterJNI.java:905)
E/AndroidRuntime( 7592): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:160)
E/AndroidRuntime( 7592): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:238)
E/AndroidRuntime( 7592): 	at com.xamdesign.safe_device.SafeDevicePlugin$1.gotLocation(SafeDevicePlugin.java:41)
E/AndroidRuntime( 7592): 	at com.xamdesign.safe_device.MockLocation.MockLocationCheck$GetLastLocation.run(MockLocationCheck.java:90)
E/AndroidRuntime( 7592): 	at java.util.TimerThread.mainLoop(Timer.java:562)
E/AndroidRuntime( 7592): 	at java.util.TimerThread.run(Timer.java:512)

```

And the whole app crashes.

```
Flutter 2.2.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b22742018b (10 days ago) • 2021-05-14 19:12:57 -0700
Engine • revision a9d88a4d18
Tools • Dart 2.13.0

```
Other checks (isJailBroken, isRealDevice, isOnExternalStorage, ...) seem to be working just fine.

**Comment by ufukhawk** [comment]
Hi  @Tomucha,

I did not encounter any problems,
Can you please run "Example" in the project and try?
and make sure you are using the latest version.

<img width="313" alt="Screen Shot 2021-05-26 at 14 57 38" src="https://user-images.githubusercontent.com/30172226/119656698-a5405b80-be33-11eb-816b-3bfc205fb08b.png">

**Comment by Tomucha** [comment]
Crashes as well.
```

💪 Running with sound null safety 💪

An Observatory debugger and profiler on Pixel 2 is available at: http://127.0.0.1:42095/x5Egj9dlrU4=/
The Flutter DevTools debugger and profiler on Pixel 2 is available at: http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A42095%2Fx5Egj9dlrU4%3D%2F
W/_device_example(13139): type=1400 audit(0.0:1325709): avc: denied { read } for name="cache" dev="dm-0" ino=16 scontext=u:r:untrusted_app:s0:c75,c259,c512,c768 tcontext=u:object_r:cache_file:s0 tclass=lnk_file permissive=0 app=com.xamdesign.safe_device_example
I/RootBeer(13139): LOOKING FOR BINARY: /data/local/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /data/local/bin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /data/local/xbin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /sbin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /su/bin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/.ext/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/failsafe/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/sd/xbin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/usr/we-need-root/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /system/xbin/su Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /cachesu Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /datasu Absent :(
I/RootBeer(13139): LOOKING FOR BINARY: /devsu Absent :(
E/AndroidRuntime(13139): FATAL EXCEPTION: Timer-0
E/AndroidRuntime(13139): Process: com.xamdesign.safe_device_example, PID: 13139
E/AndroidRuntime(13139): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: Timer-0
E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:1230)
E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.FlutterJNI.invokePlatformMessageResponseCallback(FlutterJNI.java:905)
E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:160)
E/AndroidRuntime(13139): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:238)
E/AndroidRuntime(13139): 	at com.xamdesign.safe_device.SafeDevicePlugin$1.gotLocation(SafeDevicePlugin.java:41)
E/AndroidRuntime(13139): 	at com.xamdesign.safe_device.MockLocation.MockLocationCheck$GetLastLocation.run(MockLocationCheck.java:90)
E/AndroidRuntime(13139): 	at java.util.TimerThread.mainLoop(Timer.java:562)
E/AndroidRuntime(13139): 	at java.util.TimerThread.run(Timer.java:512)
I/Process (13139): Sending signal. PID: 13139 SIG: 9
Lost connection to device.

```

It shows the example screen, then I leave it for a while (few seconds) and then it crashes. Pixel 2 phone, not rooted, no custom ROM, no special conditions or whatever.

**Comment by ufukhawk** [comment]
> Crashes as well.
> 
> ```
> 
> 💪 Running with sound null safety 💪
> 
> An Observatory debugger and profiler on Pixel 2 is available at: http://127.0.0.1:42095/x5Egj9dlrU4=/
> The Flutter DevTools debugger and profiler on Pixel 2 is available at: http://127.0.0.1:9100?uri=http%3A%2F%2F127.0.0.1%3A42095%2Fx5Egj9dlrU4%3D%2F
> W/_device_example(13139): type=1400 audit(0.0:1325709): avc: denied { read } for name="cache" dev="dm-0" ino=16 scontext=u:r:untrusted_app:s0:c75,c259,c512,c768 tcontext=u:object_r:cache_file:s0 tclass=lnk_file permissive=0 app=com.xamdesign.safe_device_example
> I/RootBeer(13139): LOOKING FOR BINARY: /data/local/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /data/local/bin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /data/local/xbin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /sbin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /su/bin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/.ext/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/bin/failsafe/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/sd/xbin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/usr/we-need-root/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /system/xbin/su Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /cachesu Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /datasu Absent :(
> I/RootBeer(13139): LOOKING FOR BINARY: /devsu Absent :(
> E/AndroidRuntime(13139): FATAL EXCEPTION: Timer-0
> E/AndroidRuntime(13139): Process: com.xamdesign.safe_device_example, PID: 13139
> E/AndroidRuntime(13139): java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: Timer-0
> E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:1230)
> E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.FlutterJNI.invokePlatformMessageResponseCallback(FlutterJNI.java:905)
> E/AndroidRuntime(13139): 	at io.flutter.embedding.engine.dart.DartMessenger$Reply.reply(DartMessenger.java:160)
> E/AndroidRuntime(13139): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.success(MethodChannel.java:238)
> E/AndroidRuntime(13139): 	at com.xamdesign.safe_device.SafeDevicePlugin$1.gotLocation(SafeDevicePlugin.java:41)
> E/AndroidRuntime(13139): 	at com.xamdesign.safe_device.MockLocation.MockLocationCheck$GetLastLocation.run(MockLocationCheck.java:90)
> E/AndroidRuntime(13139): 	at java.util.TimerThread.mainLoop(Timer.java:562)
> E/AndroidRuntime(13139): 	at java.util.TimerThread.run(Timer.java:512)
> I/Process (13139): Sending signal. PID: 13139 SIG: 9
> Lost connection to device.
> ```
> 
> It shows the example screen, then I leave it for a while (few seconds) and then it crashes. Pixel 2 phone, not rooted, no custom ROM, no special conditions or whatever.

please try 1.0.6 : https://pub.dev/packages/safe_device

**Comment by Tomucha** [comment]
Well I didn't test it yet, but if you insist :-)
